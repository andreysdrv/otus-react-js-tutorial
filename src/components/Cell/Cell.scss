.cell {
  width: 100px;
  height: 100px;
  border: 1px solid deeppink;
  box-sizing: border-box;
  transition: .2s;
  &:hover {
    background-color: darken(#f9ffb5, 10%);
    cursor: pointer;
  }
}