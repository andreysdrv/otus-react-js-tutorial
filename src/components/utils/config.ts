const FIELD_SIZE = 3;
const CELLS: number[] = [1, 2, 3, 4, 5, 6, 7, 8, 9];

export { FIELD_SIZE, CELLS };
