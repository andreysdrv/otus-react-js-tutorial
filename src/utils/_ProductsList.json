{
  "products": [
    {
      "id": 1,
      "title": "Meat",
      "completed": false
    },
    {
      "id": 2,
      "title": "Apple",
      "completed": false
    },
    {
      "id": 3,
      "title": "Milk",
      "completed": true
    }
  ]
}
