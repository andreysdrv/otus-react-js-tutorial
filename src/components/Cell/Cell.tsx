import React, { FC, MouseEventHandler } from "react";
import "./Cell.scss";

interface CellProps {
  content: number;
  handleCellClick: MouseEventHandler;
}

const Cell: FC<CellProps> = ({ content, handleCellClick }) => {
  const handleClick = (e) => handleCellClick(e.target.textContent);

  return (
    <div className="cell" onClick={handleClick}>
      {content}
    </div>
  );
};

export default Cell;
