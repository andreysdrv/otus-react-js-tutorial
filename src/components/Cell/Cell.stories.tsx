import React from "react";
import Cell from "./Cell";
import { ComponentMeta, ComponentStory } from "@storybook/react";
import { CELLS } from "../utils/config";

export default {
  component: Cell,
  title: "Cell",
  decorators: [(story) => <div style={{ padding: "1rem" }}>{story()}</div>],
} as ComponentMeta<typeof Cell>;

const Template: ComponentStory<typeof Cell> = (args) => <Cell {...args} />;

export const Clickable = Template.bind({});
Clickable.args = {
  value: CELLS[0],
};

export const Styled = Template.bind({});
Styled.args = {
  value: Cell,
  className: "cell",
};
