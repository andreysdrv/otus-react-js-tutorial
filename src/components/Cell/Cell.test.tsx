import React from "react";
import { render, screen } from "@testing-library/react";
import Cell from "./Cell";
import userEvent from "@testing-library/user-event";
import { CELLS } from "../utils/config";

describe("Cell snapshot tests", () => {
  it("base snapshot", () => {
    const { container } = render(
      <Cell content={1} handleCellClick={() => console.log(1)} />
    );
    expect(container).toMatchSnapshot();
  });

  it("some data snapshot", () => {
    const { container } = render(
      <Cell content={1} handleCellClick={() => console.log(1)} />
    );
    expect(container).toMatchSnapshot();
  });

  it("handleCellClick", async () => {
    const handleCellClick = jest.fn();
    const user = userEvent.setup();

    render(<Cell content={CELLS[0]} handleCellClick={handleCellClick} />);

    const cell = screen.getByText(CELLS[0]);
    await user.click(cell);

    expect(handleCellClick).toHaveBeenCalled();
  });
});
