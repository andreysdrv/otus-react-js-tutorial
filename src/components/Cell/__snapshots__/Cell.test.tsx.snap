// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cell snapshot tests base snapshot 1`] = `
<div>
  <div
    class="cell"
  >
    1
  </div>
</div>
`;

exports[`Cell snapshot tests some data snapshot 1`] = `
<div>
  <div
    class="cell"
  >
    1
  </div>
</div>
`;
